components/base/wheeled/wheeled_base.go

```diff
@@ -97,6 +100,9 @@ type wheeledBase struct {
 
 	opMgr  operation.SingleOperationManager
 	logger golog.Logger
+
+	// TODO(rb): this is used for building the model and should eventually move out of this struct
+	cfg config.Component
```

Comment:
[minor] it seems like it could be moved to the `CreateWheeledBase` function using your helper. We pass the config through there to create the driver objects. You could check if the frame config exists there and wrap it.

Reply:
Not sure I completely follow.  You're saying instead of doing this, make the model in the normal wheeledBase constructor and tore the model in the wheeledBase?

Reply:
don't store the whole `cfg` in the base struct, you can either : make the `model` store the model in the wheeled base  as `wheeledBase.model` and access it through there for your kinematic base; or- pass the `cfg` or `cfg.Frame` as another variable into your helper function to create the `kinematicWheeledBase` and access frame information there. Or some other way if you find it. We should not store the `cfg` itself in our structs. We'd rather keep configuration-related information and logic in the part of the code that constructs, reconfigure, and destroys resources rather than all throughout this resource.

Reply:
Cool I like the first option a lot better thanks